kubernetes/schema_container.go

```diff
@@ -384,6 +384,7 @@ func containerFields(isUpdatable, isInitContainer bool) map[string]*schema.Schem
 		},
 		"image": {
 			Type:        schema.TypeString,
+			ForceNew:    !isUpdatable,
```

Comment:
Only added `ForceNew:    !isUpdatable` here as that's what made me notice the issue, but it should probably be generalized to other fields.